### ISSUE TITLE
String's special characters not visible when received from R-API into Android app.

### ISSUE THREAD

**Issue by Alfiva** [universAAL/remote#475]
_Originally Opened: @Alfiva (2015-09-02 18:27:47_)
_Originally Closed: 2015-09-02 18:40:35_

Consider an Android device running uAAL, connected through R-API to a uAAL server instance. When the server initiates a communication (e.g. requests a S Call), String values are correctly received by the app addressed by the call. But when it is the app that sends a S Call addressed at another app in the server, the Strings received from the response will display the special characters (e.g: á) incorrectly.

--

From: _this issue has been automatically imported from our old issue tracker_

**Comment by Alfiva** [comment]
_Original author: @Alfiva (2015-09-02 18:31:58)_

This was originally posted in nativeandroid but it is actually R-API's fault. Moved here.

**Comment by Alfiva** [comment]
_Original author: @Alfiva (2015-09-02 18:40:35)_

The problem was that when sending ServiceResponses from the ri.api.manager, I forgot to set UTF-8 as the charset, while I had set it correctly everywhere else. It is now fixed. The reason it worked OK when pushing calls from the server is because in that case it went through GCM, while in the other direction it was a direct HTTP response.

I have also updated the uAAL android app because it was missing setting UTF-8 when receiving this info, although that wasn't the problem.
